src/main/java/org/jspecify/annotations/Nullable.java

```diff
@@ -42,88 +42,95 @@
+ *       method {@code @Nullable E eOrNull()}. Then, whether a variable {@code foo} is of type
  *       {@code Foo<String>} or {@code Foo<@Nullable String>}, the expression {@code foo.eOrNull()}
- *       is nullable either way. Using {@code @Nullable E} in this way is called "nullable
- *       projection" (<a href="NonNull.html#projection">non-null projection</a> is likewise
- *       supported, but less commonly useful).
+ *       is nullable either way. Using {@code @Nullable E} in this way is called <i>nullable
+ *       projection</i> (the more rarely needed <a href="NonNull.html#projection">non-null
+ *       projection</a> is also supported).
+ *   <li>On an <b>array type</b>: The nullness of the array and of its components can both be
+ *       indicated separately. For example, in null-marked code, {@code @Nullable String[]} is a
+ *       non-null array of nullable strings, whereas {@code String @Nullable []} is a nullable array
+ *       of non-null strings. Note that arrays are treated as covariant: {@code String[]} (array of
+ *       non-null strings) is assignable to {@code @Nullable String[]} (array of nullable strings).
  *   <li>On a <b>nested type</b>: In most examples above, in place of {@code String} we might use a
  *       nested type such as {@code Map.Entry}. The Java syntax for annotating such a type as
  *       nullable looks like {@code Map.@Nullable Entry}.
+ *   <li>On a parameter declaration of variable arity ("varargs"): Use {@code @Nullable String...}
+ *       to indicate that the individual strings may be null.
  *   <li>On a <b>record component</b>: As expected, {@code @Nullable} here applies equally to the
- *       corresponding parameter type of the canonical constructor, and to the return type of a
- *       generated accessor method as well. If an explicit accessor method is provided for this
- *       record component, it must still be annotated explicitly. Any non-null components should be
- *       checked (for example using {@link java.util.Objects#requireNonNull}) in a <a
- *       href="https://docs.oracle.com/en/java/javase/19/language/records.html">compact
- *       constructor</a>.
+ *       corresponding parameter type of the canonical constructor, and to the return type of the
+ *       corresponding accessor method, if these are actually generated. If the constructor
+ *       signature or accessor method is written explicitly, it must still be annotated explicitly.
  * </ul>
  *
- * <h2 id="applicability">Where it is applicable</h2>
+ * <!-- TODO: move up -->
+ * <h2 id="applicability">Where it is not applicable</h2>
  *
  * <p>This annotation and {@link NonNull} are applicable to any <a
  * href="https://github.com/jspecify/jspecify/wiki/type-usages">type usage</a> <b>except</b> the
  * following cases, where they have no defined meaning:
  *
  * <ul>
+ *   <li>On a <b>local variable</b> declaration. According to JSpecicy, the nullness of a local
```

Comment:
Do we really need "according to JSpecify" here? I feel like that's implied, or else we'd have to sprinkle it all over the place.